ckanext/search/tests/test_filters.py

```diff
@@ -125,6 +234,77 @@ def test_filters_list_of_filters_combined_as_or():
         ],
     )
 
+
+def test_filters_test_nested_operators_combined():
+
+    filters = {
+        "field1": "value1",
+        "$and": [
+            {"field2": {"gte": 5, "lte": 7}},
+            {
+                "$and": [
+                    {"field3": "value3"},
+                    {"field4": "value4"},
+                ]
+            },
+        ],
+        "$or": [
+            {"field5": "value5"},
+            {
+                "$or": [
+                    {"field6": "value6"},
+                    {"field7": "value7"},
+                ]
+            },
+        ],
+    }
+    result = query_filters_validator(filters)
+    assert result == FilterOp(
+        field=None,
+        op="$and",
+        value=[
+            FilterOp(field="field1", op="eq", value="value1"),
+            FilterOp(field="field2", op="gte", value=5),
+            FilterOp(field="field2", op="lte", value=7),
+            FilterOp(field="field3", op="eq", value="value3"),
+            FilterOp(field="field4", op="eq", value="value4"),
+            FilterOp(
+                field=None,
+                op="$or",
+                value=[
+                    FilterOp(field="field5", op="eq", value="value5"),
+                    FilterOp(field="field6", op="eq", value="value6"),
+                    FilterOp(field="field7", op="eq", value="value7"),
+                ],
+            ),
+        ],
+    )
+
+
+def test_filters_list_of_filters_with_or_operator():
+    filters = [
+        {"field1": "value1"},
+        {"field2": "value2"},
+        {
+            "$or": [
+                {"field3": "value3"},
+                {"field4": "value4"},
+            ]
+        },
+    ]
+    result = query_filters_validator(filters)
+    assert result == FilterOp(
+        field=None,
+        op="$or",
+        value=[
+            FilterOp(field="field1", op="eq", value="value1"),
+            FilterOp(field="field2", op="eq", value="value2"),
+            FilterOp(field="field3", op="eq", value="value3"),
+            FilterOp(field="field4", op="eq", value="value4"),
+        ],
+    )
+
+
 # TODO: what do we expect here?
 # def test_filters_single_or():
 #     filters = {
```

Comment:
Single-value or/and are either the same as unwrapping the inner operator (behaving like `all()`/`any()`) or an error if we want to be strict with this case too.